File: KRVRJS/src/models/Material.js
class Material {
  constructor() {
    this._id = Date.now();
    this._x = 0;
    this._y = 0;
    this._z = 0;
    this._width = 0;
    this._height = 0;
    this._depth = 0;
  }

  get x() {
    return this._x;
  }

  set x(x) {
    this._x = x;
  }

  get y() {
    return this._y;
  }

  set y(y) {
    this._y = y;
  }

  get z() {
    return this._z;
  }

  set z(z) {
    this._z = z;
  }

  get width() {
    return this._width;
  }

  set width(width) {
    this._width = width;
  }

  get height() {
    return this._height;
  }

  set height(height) {
    this._height = height;
  }

  get depth() {
    return this._depth;
  }

  set depth(depth) {
    this._depth = depth;
  }
}

export default Material;

File: KRVRSW/model/Material.py
import json

class Material:
    def __init__(self):
        self.x = 0
        self.y = 0
        self.z = 0
        self.width = 0
        self.height = 0
        self.depth = 0

    def toJSON(self):
        return json.dumps(self, default=lambda o: o.__dict__)

class MaterialEncoder(json.JSONEncoder):
    def default(self, obj):
        if isinstance(obj, Material):
            return obj.__dict__
        return json.JSONEncoder.default(self, obj)
